app/tasks/s3.py

```diff
@@ -0,0 +1,70 @@
+from __future__ import \
+    unicode_literals, \
+    absolute_import
+
+from . import celery
+from flask import current_app
+from io import BytesIO
+from os import getenv
+from app.api.services import brief_responses_service
+from app.models import BriefResponse
+from werkzeug.utils import secure_filename
+import boto3
+import botocore
+import zipfile
+
+
+class CreateResumesZipException(Exception):
+    """Raised when the resume zip fails to create."""
+
+
+@celery.task
+def create_resumes_zip(brief_id):
+    BUCKET_NAME = current_app.config.get('S3_BUCKET_NAME')
+    s3 = boto3.resource(
+        's3',
+        region_name=getenv('AWS_REGION'),
+        aws_access_key_id=getenv('AWS_ACCESS_KEY_ID'),
+        aws_secret_access_key=getenv('AWS_SECRET_ACCESS_KEY')
+    )
+    bucket = s3.Bucket(BUCKET_NAME)
+
+    files = []
+    brief_responses = brief_responses_service.get_brief_responses(brief_id, None)
+    for brief_response_simple in brief_responses:
+        brief_response = BriefResponse.query.get(brief_response_simple['id'])
+        attachments = brief_response.data.get('attachedDocumentURL', [])
+        for attachment in attachments:
+            files.append({
+                'key': 'supplier-{}/{}'.format(brief_response.supplier_code, attachment),
+                'zip_name': 'brief-{}/{}/{}'.format(
+                    brief_id,
+                    secure_filename(brief_response.supplier.name),
+                    attachment
+                )
+            })
+
+    if not files:
+        raise CreateResumesZipException('The brief id "{}" did not have any attachments'.format(brief_id))
+
+    archive = BytesIO()
```

Comment:
so the zip is created in memory before upload to S3?

Reply:
correct yes - it will be the celery worker app doing it mostly, based on how it is currently tasked (via the closed brief email which is a scheduled task), so it shouldn't burden the main API app's memory.